Engine/typesystem_engine.xml

```diff
@@ -1506,14 +1509,7 @@
 
                     for (int j = 0; j &lt; subSize; ++j) {
                         PyObject* pyString = PyList_GET_ITEM(subList,j);
-                        if ( PyString_Check(pyString) ) {
-                            char* buf = PyString_AsString(pyString);
-                            if (buf) {
-                                std::string ret;
-                                ret.append(buf);
-                                rowVec[j] = ret;
-                                }
-                        } else if (PyUnicode_Check(pyString) ) {
+                        if (PyUnicode_Check(pyString) ) {
```

Comment:
instead of removing,
- replace PyString_Check by PyBytes_Check
- replace PyString_AsString by PyObject_Bytes
- put the if ( PyBytes_Check(pyString) ) { section after the if (PyUnicode_Check(pyString) section

Reply:
(this is to keep python2 compatibility)

Reply:
Gonna follow [PEP-393](https://www.python.org/dev/peps/pep-0393/) by adapting what you've suggested. May be better for Shiboken to be smart enough for `#if PY_MAJOR_VERSION > 3`/`#else`.

Reply:
I think the PyBytes_Check works both in python2 and 3 (PyString_Check disappeared from Python3), so that makes the code 2/3 compatible. Are you suggesting it should be done differently? There is similar code at several places in Natron

Reply:
Luckily enough one call to `PyUnicode_Check` is needed to check if the type is suitable for Python strings, according to the [official docs](https://docs.python.org/3/library/stdtypes.html#str). Although the `PyBytes_Check` method works too.